src/snowflake/connector/file_transfer_agent.py

```diff
@@ -1017,6 +1017,8 @@ def _init_file_metadata(self) -> None:
                         )
                     )
         elif self._command_type == CMD_TYPE_DOWNLOAD:
+            existing_basenames = set()
```

Comment:
small and biased NIT: we could use a counter/dictionary. And duplicate base names are the entries with >1 occurrence.